sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.institutions.controller.ts

```diff
@@ -139,4 +161,85 @@ export class ScholasticStandingInstitutionsController extends BaseController {
       userToken.authorizations.getLocationsIds(),
     );
   }
+
+  /**
+   * Process a text file with applications to be withdrawn.
+   * @param file text file content with all information needed to withdraw applications.
+   * @param validationOnly if true, will execute all validations and return the
+   * errors and warnings. These validations are the same executed during the
+   * final creation process. If not present or false, the file will be processed
+   * and the records will be inserted.
+   * @returns when successfully executed, the list of all withdrawn application ids.
+   * When an error happen it will return all the records (with the error) and
+   * also a user friendly description of the errors to be fixed.
+   */
+  @ApiBadRequestResponse({
+    description:
+      "Error while parsing text file or " +
+      "one or more text data fields received are not in the correct format or " +
+      "there are no records to be imported.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Application withdrawal has invalid data or " +
+      "some error happen with one or more application withdrawal and the entire process was aborted.",
+  })
+  @IsInstitutionAdmin()
+  @IsBCPublicInstitution()
+  @Post("application-bulk-withdrawal")
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(
+        MAX_UPLOAD_FILES,
+        APPLICATION_BULK_WITHDRAWAL_MAX_UPLOAD_PARTS,
+        APPLICATION_BULK_WITHDRAWAL_UPLOAD_MAX_FILE_SIZE,
+      ),
+      fileFilter: textFileFilter,
+    }),
+  )
+  async bulkWithdrawal(
+    @UploadedFile() file: Express.Multer.File,
```

Comment:
Are we not validating the MIME Type to be `text/plain` ?

Reply:
Yes we are validating using the textFileFilter in the UseInterceptors 4 lines before.

Reply:
Correct me if wrong, the textFileFilter is checking for file extension only, not the mime type.  Should we rather check for mime type?